### PR TITLE
Plot_analyze_with_lightkurve

### PR DESCRIPTION
**Pull request by jrka**

### CODE REVIEW

File: notebooks/MAST/plot_analyze_with_lightkurve/plot_analyze_with_lightkurve.ipynb [hunk 1]
```diff
@@ -0,0 +1,953 @@
+{
```
Comment (kelle) [inline notebooks/MAST/plot_analyze_with_lightkurve/plot_analyze_with_lightkurve.ipynb:1]: <p>Let's discuss how to install extra packages in #11</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ScienceBetter/notebooks-stsci/pull/13/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Comment (kelle) [inline notebooks/MAST/plot_analyze_with_lightkurve/plot_analyze_with_lightkurve.ipynb:1]: <p>move solution to dedicated notebook.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ScienceBetter/notebooks-stsci/pull/13/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='22'/>

**Comment by kelle** [comment]
This also looks great to me and is ready for review.

File: notebooks/MAST/plot_analyze_with_lightkurve/plot_analyze_with_lightkurve.ipynb (hunk 1 above)
Comment (hmlewis-astro) [inline notebooks/MAST/plot_analyze_with_lightkurve/plot_analyze_with_lightkurve.ipynb:1]: <p>Make sure any changes applied to Part 1 are applied here as well (small spelling and rewording comments).</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ScienceBetter/notebooks-stsci/pull/13/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (hmlewis-astro) [inline notebooks/MAST/plot_analyze_with_lightkurve/plot_analyze_with_lightkurve.ipynb:1]: <p>Here (and in Part 1), may be useful to refer to the star by its TIC ID in addition to giving the coordinates, e.g., "TIC 373283207 (RA = 2.238638, Dec = 73.690475)"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ScienceBetter/notebooks-stsci/pull/13/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

Comment (hmlewis-astro) [inline notebooks/MAST/plot_analyze_with_lightkurve/plot_analyze_with_lightkurve.ipynb:1]: <p>In the second paragraph, you say "From plotting the two types of fluxes..." but only explain one of the types (KSPSAP_FLUX). Need to add a bit about SAP_FLUX.</p><p> </p><p>Also instead of "Such long timescale variation is intrinsic stellar variability" I would rephrase slightly: "This long timescale variability is caused by intrinsic stellar variations" or something similar.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ScienceBetter/notebooks-stsci/pull/13/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='15'/>

Comment (hmlewis-astro) [inline notebooks/MAST/plot_analyze_with_lightkurve/plot_analyze_with_lightkurve.ipynb:1]: <p>Small grammatical fix: "This package is <strong>built</strong> specifically..."</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ScienceBetter/notebooks-stsci/pull/13/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='23'/>

Comment (hmlewis-astro) [inline notebooks/MAST/plot_analyze_with_lightkurve/plot_analyze_with_lightkurve.ipynb:1]: <p>Second sentence, suggested rewording: "This also helps us to confirm that by default, the lightkurve plotting function uses the sap_flux on the flux axis, rather than the detrended kspsap_flux; the sap_flux can be used for studies of stellar variability, but these signals need to be removed from the light curve for detailed studies of eclipses and transits."</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ScienceBetter/notebooks-stsci/pull/13/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='29'/>

Comment (hmlewis-astro) [inline notebooks/MAST/plot_analyze_with_lightkurve/plot_analyze_with_lightkurve.ipynb:1]: <p>I would recommend putting these plotting functions in two different cells, with the note "We must be sure to specify that we want..." preceding the second call to plot (i.e., with column='kspsap_flux').</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ScienceBetter/notebooks-stsci/pull/13/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='30'/>

Comment (hmlewis-astro) [inline notebooks/MAST/plot_analyze_with_lightkurve/plot_analyze_with_lightkurve.ipynb:1]: <p>Suggested rewording of how the data were collected: "Examining the full amount of data available, we can see that the target of interest was observed in two TESS Sectors back-to-back (Sectors 18 and 19) and then again in Sectors 24 and 25, with a gap of approximately 100 days between the end of Sector 19 and the beginning of Sector 24."</p><p> </p><p>Spelling correction: "Data <strong>Gaps</strong> and Quality Flags"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ScienceBetter/notebooks-stsci/pull/13/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='33'/>

Comment (hmlewis-astro) [inline notebooks/MAST/plot_analyze_with_lightkurve/plot_analyze_with_lightkurve.ipynb:1]: <p>I like looking at the flagged data, but we definitely don't want to use the KeplerQualityFlags.decode here. The QLP flags do not correspond to the Kepler flags. Instead, you should refer to the QLP Data Production Description Doc (https://archive.stsci.edu/hlsps/qlp/hlsp_qlp_tess_ffi_all_tess_v1_data-prod-desc.pdf) for what the 4096 quality flag means.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ScienceBetter/notebooks-stsci/pull/13/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='34'/>

Comment (hmlewis-astro) [inline notebooks/MAST/plot_analyze_with_lightkurve/plot_analyze_with_lightkurve.ipynb:1]: <p>I think this is a good choice (excluding anything flux measurements that don't have quality == 0)!</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ScienceBetter/notebooks-stsci/pull/13/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='35'/>

Comment (hmlewis-astro) [inline notebooks/MAST/plot_analyze_with_lightkurve/plot_analyze_with_lightkurve.ipynb:1]: <p>I wouldn't call these "notably dips"; maybe just "low flux measurements"?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ScienceBetter/notebooks-stsci/pull/13/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='37'/>

Comment (hmlewis-astro) [inline notebooks/MAST/plot_analyze_with_lightkurve/plot_analyze_with_lightkurve.ipynb:1]: <p>Yes, I think you definitely want to filter out the data points that don't have quality == 0.</p><p> </p><p>Could you just do something like this?:</p><pre class="ql-syntax" spellcheck="false">lc_stitched = lc_stitched[lc_stitched['quality']==0]
lc_stitched.flux=lc_stitched['kspsap_flux']
lc_stitched.flux_err=lc_stitched['kspsap_flux_err']
lc_stitched.meta['FLUX_ORIGIN']='kspsap_flux'
</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ScienceBetter/notebooks-stsci/pull/13/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='40'/>

Comment (hmlewis-astro) [inline notebooks/MAST/plot_analyze_with_lightkurve/plot_analyze_with_lightkurve.ipynb:1]: <p>Would definitely like to see what this plot looks like without the included low quality flux measurements... Might not change at all, but would be great to confirm.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ScienceBetter/notebooks-stsci/pull/13/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='43'/>

Comment (hmlewis-astro) [inline notebooks/MAST/plot_analyze_with_lightkurve/plot_analyze_with_lightkurve.ipynb:1]: <p>Rewording suggestion: "... in fact, there is no clear periodicity or repeating signal in the light curve of our target."</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ScienceBetter/notebooks-stsci/pull/13/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='44'/>

Comment (hmlewis-astro) [inline notebooks/MAST/plot_analyze_with_lightkurve/plot_analyze_with_lightkurve.ipynb:1]: <p>Should clarify, "there are many (nearly 5000!) candidate binary systems in the Price-Whelan et al. 2018 catalog that may have eclipses..." There are many more candidates in other catalogs too!</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ScienceBetter/notebooks-stsci/pull/13/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='46'/>

Comment (jrka) [reply notebooks/MAST/plot_analyze_with_lightkurve/plot_analyze_with_lightkurve.ipynb:1]: <p>I may just be having a silly moment, but what is TIC ID? For now I put the APOGEE ID from the catalog, explained as "TMASS-STYLE object name (2MHHMMSSss+DDMMSSs) (APOGEE_ID)"</p>

Comment (jrka) [inline notebooks/MAST/plot_analyze_with_lightkurve/plot_analyze_with_lightkurve.ipynb:1]: <p>Note for me to replace "lightcurve" everywhere with "light curve" as per comments on pull request 12.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ScienceBetter/notebooks-stsci/pull/13/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (jrka) [inline notebooks/MAST/plot_analyze_with_lightkurve/plot_analyze_with_lightkurve.ipynb:1]: <p>Note to myself to change this to use astropy coordinates instead of string; like in part 1. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ScienceBetter/notebooks-stsci/pull/13/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

Comment (jrka) [reply notebooks/MAST/plot_analyze_with_lightkurve/plot_analyze_with_lightkurve.ipynb:1]: <p>Oh - I found it farther down in the tutorial ;-) It's on the TESS spectra. However, at this point in the tutorial, the user knows there are TESS spectra but hasn't yet accessed them. Or was there an earlier place where this ID number would have been known? </p>

Comment (jrka) [reply notebooks/MAST/plot_analyze_with_lightkurve/plot_analyze_with_lightkurve.ipynb:1]: <p>I assume the pdf listed has a typo and that low precision points should be 4096 not 4196?</p>

Comment (jrka) [reply notebooks/MAST/plot_analyze_with_lightkurve/plot_analyze_with_lightkurve.ipynb:1]: <p>Such an easy fix and it worked :-) </p>

Comment (hmlewis-astro) [reply notebooks/MAST/plot_analyze_with_lightkurve/plot_analyze_with_lightkurve.ipynb:1]: <p>APOGEE ID is good too! Especially since the TESS *light curves haven't been accessed at this point, and it might not be clear what the TIC ID is yet.</p><p> </p><p>The TIC ID <em>is</em> given in Part 1 of the notebook ("target_name" in "masttbl" in the final cell) and is likely also given by one of the columns in your "obs_tbl" in the next cell. Up to you if you want to reference it here; I think APOGEE ID is perfect.</p>

Comment (hmlewis-astro) [reply notebooks/MAST/plot_analyze_with_lightkurve/plot_analyze_with_lightkurve.ipynb:1]: <p>Ah, good catch. Yes, that is almost certainly a typo.</p>

File: notebooks/MAST/plot_analyze_with_lightkurve/plot_analyze_with_lightkurve.ipynb [hunk 2]
```diff
@@ -0,0 +1,987 @@
...
```
Comment (jrka) [inline notebooks/MAST/plot_analyze_with_lightkurve/plot_analyze_with_lightkurve.ipynb:1]: <p>Apparently there is a built-in way to do this:</p><p><a href="https://docs.lightkurve.org/reference/api/lightkurve.LightCurve.select_flux.html" target="_blank">https://docs.lightkurve.org/reference/api/lightkurve.LightCurve.select_flux.html</a> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ScienceBetter/notebooks-stsci/pull/13/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='41'/>
